assets/scripts/app.js

```diff
@@ -0,0 +1,216 @@
+const ATTACK_VALUE = 10;
+const STRONG_ATTACK_VALUE = 17;
+const MONSTER_ATTACK_VALUE = 14;
+const HEAL_VALUE = 20;
+const MODE_ATTACK = 'ATTACK';
+const MODE_STRONG_ATTACK = 'STRONG_ATTACK';
+const LOG_EVENT_PLAYER_ATTACK = 'PLAYER_ATTACK';
+const LOG_EVENT_PLAYER_STRONG_ATTACK = 'PLAYER_STRONG_ATTACK';
+const LOG_EVENT_MONSTER_ATTACK = 'MONSTER_ATTACK';
+const LOG_EVENT_HEAL = 'HEAL';
+const LOG_EVENT_GAME_OVER = 'GAME_OVER';
+const battleLog = [];
+
+
+function writeToLog (ev, val, playerHealth, monsterHelath){
+    logEntry = {
+        event: ev,
+        value: val,
+        finalPlayerHealth: playerHealth,
+        finalMonsterHealth: monsterHelath
+    };
+
+    switch (ev){
+        case LOG_EVENT_MONSTER_ATTACK:
+            logEntry.target = 'Player';
+            break;
+        case LOG_EVENT_PLAYER_ATTACK:
+            logEntry.target = 'Player';
+            break;
+        case LOG_EVENT_PLAYER_STRONG_ATTACK:
+            logEntry = {
+                event: ev,
+                value: val,
+                target: 'Monster',
+                finalPlayerHealth: playerHealth,
+                finalMonsterHealth: monsterHelath
+            };
+            break;
+        case LOG_EVENT_HEAL:
+            logEntry = {
+                event: ev,
+                value: val,
+                target: 'Player',
+                finalPlayerHealth: playerHealth,
+                finalMonsterHealth: monsterHelath
+            };
+            break;
+        case LOG_EVENT_GAME_OVER:
+            logEntry = {
+                event: ev,
+                value: val,
+                finalPlayerHealth: playerHealth,
+                finalMonsterHealth: monsterHelath
+            };
+            break;
+        default:
+            logEntry = {};
+    }
+
+    battleLog.push(logEntry);
+  
+}
+
+
+function getMaxLifeValue(){
+    const enteredValue = prompt('Choose a Maximum life', '100');
+    const parsedValue = parseInt(enteredValue);
+
+    if (isNaN(parsedValue) || parsedValue <=0){
+        throw{message: 'Invalid Input, enter a number!'};
+    }
+    return parsedValue;
+}
+
+try {
+    choosenMaxLife = getMaxLifeValue();
+} catch (error) {
+    console.error;
+    choosenMaxLife = 100;
+    alert('Invalid input, default value 100 was selected ');
+}
+
+let currentMonsterHealth = choosenMaxLife;
+let currentPlayerHealth = choosenMaxLife;
+let hasBonus = true;
+
+
+adjustHealthBars(choosenMaxLife);
+
+
+function reset(){
+    currentMonsterHealth = choosenMaxLife;
+    currentPlayerHealth = choosenMaxLife;
+    resetGame(choosenMaxLife);
+}
+
+
+function endRound(){
```

Comment:
`u can use here arrow function`

Reply:
I purposely didn't used it beacause I wasn't aware of uit back then @syedabdullahali